### PR TITLE
v3: jsdoc to docusaurus

### PR DESCRIPTION
**Pull request by malangfox**
## Details
Documentation updated to docusaurus from jsdoc.
This documentation includes installation giude, api docs and demos.
You can run `npm start` at egjs-axes/demo to check the documentation.

### CODE REVIEW

File: demo/docs/tutorials/Custom Inputtype.mdx [hunk 1]
```diff
@@ -0,0 +1,162 @@
+---
+title: Creating a custom InputType
+id: custom-inputtype
+slug: /custom-inputtype
```
Comment (WoodNeck) [inline demo/docs/tutorials/Custom Inputtype.mdx:4]: I'd recommend using custom-input-type instead, same for the file name(Custom Input Type)

File: demo/docs/tutorials/Custom Inputtype.mdx [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+>	element: HTMLElement;
+>	mapAxes(axes: string[]);
+>	connect(observer: IInputTypeObserver): IInputType;
+>	disconnect();
+>	destroy();
+>}
+>```
+
+### Method
+#### 1. Constrouctor
```
Comment (WoodNeck) [inline demo/docs/tutorials/Custom Inputtype.mdx:98]: Here's a typo, should be `constructor`

File: demo/docs/tutorials/Custom Inputtype.mdx [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+       throttle: 100
+    }, ...options
+  };
+  // ...
+}
+```
+
+#### 2. connect(observer: IInputTypeObserver): IInputType `required`
+Register an event that can receive user input.
+It is called by eg.Axes when [axes.connect] (https://naver.github.io/egjs-axes/release/latest/doc/eg.Axes.html#connect) is called.
```
Comment (WoodNeck) [inline demo/docs/tutorials/Custom Inputtype.mdx:121]: I wonder this works correctly, shouldn't there be no space between [] and () in markdown?

File: demo/src/pages/demos/3dcarousel.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+  });
+
+  return (
+    <div className="demobox" codepen="3dcarousel">
+      <p>You can create a carousel by controlling the animation.</p>
+      <div id="carouselWrapper">
+        <div className="list_container">
+          <div id="carousel" style={{ transform: `translateZ(-253px)` }}>
+            <figure style={{ transform: `rotateY(0deg) translateZ(253px)` }}>
+              <div id="list_cd" style={{ backgroundImage: "url("+require("@site/static/img/demos/3dcarousel/music1-min.jpg").default+")" }}>
```
Comment (WoodNeck) [inline demo/src/pages/demos/3dcarousel.tsx:45]: Using backtick (\`) looks wrong here and there.
transform's value should be covered with \", and backgroundImage should rather use backtick.

File: src/AnimationManager.ts [hunk 5]
```diff
@@ -1,377 +1,468 @@
...
-				getDecimalPlace(options.range[0]),
-				getDecimalPlace(options.range[1]),
-				getDecimalPlace(val)));
-		}
-
-		return minRoundUnit || roundUnit;
-	}
+  public itm: InterruptManager;
+  public em: EventManager;
+  public axm: AxisManager;
```
Comment (WoodNeck) [inline src/AnimationManager.ts:35]: I always thought this is quite a bad naming for public properties, how about changing it?
I think it should be done before releasing v3

**Review by WoodNeck — APPROVED** [review]
LGTM, but would like to check the actual demo page!

File: demo/docs/tutorials/Custom Inputtype.mdx (hunk 3 above)
Comment (malangfox) [reply demo/docs/tutorials/Custom Inputtype.mdx:121]: All documents with the same error have been fixed, including the demo documentation and axes github wiki.

File: src/AnimationManager.ts (hunk 5 above)
Comment (malangfox) [reply src/AnimationManager.ts:35]: Also, is it fine to change the type names like IInputType, IInputTypeObserver to InputType and InputTypeObserver? There was an discussion before and I think it's time to change these names.

**Comment by WoodNeck** [comment]
Please add the missing icons file in `demo/static/icon`

**Comment by daybrush** [comment]
How about removing react-axes because it is not managed now and will be reorganized later?

File: demo/src/pages/demos/character.tsx [hunk 6]
```diff
@@ -0,0 +1,284 @@
+/* eslint-disable max-classes-per-file */
+import React, { Component } from "react";
+
+import Axes, {
+  PanInput,
+  MoveKeyInput,
+  PinchInput,
+  WheelInput,
+} from "../../../../packages/react/src";
```
Comment (daybrush) [inline demo/src/pages/demos/character.tsx:9]: It is recommended not to use react-axes.

**Review by daybrush — APPROVED** [review]
LGTM

File: demo/src/pages/demos/character.tsx (hunk 6 above)
Comment (qpitlove) [reply demo/src/pages/demos/character.tsx:9]: Let's add a new demo in the next patch update

**Review by qpitlove — APPROVED** [review]

**Pull request merged**